thali/thalireplicationmanager.js

```diff
@@ -192,11 +202,44 @@ function muxServerBridge(tcpEndpointServerPort) {
   var serverPlex = multiplex({}, function(stream, id) {
     var clientSocket = net.createConnection({port: tcpEndpointServerPort});
     stream.pipe(clientSocket).pipe(stream);
+
+    clientSocket.on('end', function () {
+      clientSocket.destroy();
```

Comment:
If we got an end event doesn't that mean the socket is already destroyed? Any danger from destroying a socket twice?